src/common.speech/ServiceRecognizerBase.ts

```diff
@@ -159,13 +174,55 @@ export abstract class ServiceRecognizerBase implements IDisposable {
             }
         });
 
-        const phraseDetection: PhraseDetection = {};
+        if (this.privEnableSpeakerId) {
+            this.privDiarizationSessionId = createNoDashGuid();
+            // this.setSpeakerDiarizationJson(phraseDetection);
+        }
 
-        if (recognizerConfig.autoDetectSourceLanguages !== undefined) {
-            const sourceLanguages: string[] = recognizerConfig.autoDetectSourceLanguages.split(",");
+        this.setLanguageIdJson();
+        this.setOutputDetailLevelJson();
+        this.setSpeakerDiarizationJson();
```

Comment:
This call should also be in the subclass constructor, since it depends on first reading the config value into privEnableSpeakerId. The method definition should probably also go in ConversationTranscriptionServiceRecognizer.